discretize/base/base_mesh.py

```diff
@@ -33,103 +31,164 @@ class BaseMesh(properties.HasProperties, InterfaceMixins):
         "nFz": "n_faces_z",
         "nF": "n_faces",
         "vnF": "n_faces_per_direction",
+        "vnC": "shape_cells"
     }
-
-    # Properties
-    _n = properties.Tuple(
-        "Tuple of number of cells in each direction (dim, )",
-        prop=properties.Integer(
-            "Number of cells along a particular direction", cast=True, min=1
-        ),
-        min_length=1,
-        max_length=3,
-        coerce=True,
-        required=True,
-    )
-
-    origin = properties.Array(
-        "origin of the mesh (dim, )",
-        dtype=(float, int),
-        shape=("*",),
-        required=True,
-    )
+    _items = {'shape_cells', 'origin', 'orientation', 'reference_system'}
```

Comment:
This looks good! A couple questions / thoughts on naming: 
- do we want to use a name that somehow indicates these are the properties that are serialized? e.g. `_serialized_properties` or something that indicates these are the set of properties that are necessary for re-creating the class? I don't feel strongly about this, just trying to explore if there is something that might be a touch more descriptive. 

- `reference_system` --> `coordinate_system` ?

Reply:
What is the reason for `_items` being a dict, given they are only keys?

Reply:
(...wouldn't it be better an immutable tuple than a mutable dict?)

Reply:
This is not a `dict`, it's actually a `set`! (A set can only have unique entries)

Reply:
Oh goodness me, of course, it was clearly too late yesterday :facepalm: